### PR TITLE
Integrate the mixer in the deployment

### PR DESCRIPTION
**Pull request by andraxylia**

### CODE REVIEW

File: test/integration/app-proxy.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        args:
+          - --port
+          - "{{.port1}}"
+        ports:
+        - containerPort: {{.port1}}
+      - name: proxy
+        image: gcr.io/istio-testing/proxy
+        args:
+          - -l 2
+          - -m mixer
```
Comment (kyessenov) [inline test/integration/app-proxy.yaml.tmpl]: --logtostderr

Comment (kyessenov) [inline test/integration/app-proxy.yaml.tmpl]: also this file is not used by anything?

File: test/integration/driver.go [hunk 2]
```diff
@@ -122,75 +129,33 @@ func main() {
...
+		w = bufio.NewWriter(f)
 	}
 
-	if err := write("test/integration/external-services.yaml.tmpl", map[string]string{
+	if err:= write("test/integration/"+svcType+".yaml.tmpl", map[string]string{
 		"hub":       hub,
 		"tag":       tag,
 		"namespace": namespace,
+		"name":      svcName,
+		"port1":     "8080",
```
Comment (kyessenov) [inline test/integration/driver.go]: we need 8080/80 and 80/8080 combination - there are a lot of edge cases and this test is reducing coverage.

File: test/integration/app-proxy.yaml.tmpl (hunk 1 above)
Comment (andraxylia) [inline test/integration/app-proxy.yaml.tmpl]: Ok, we can pass the variable parts (port1, port2) as parameters.

The app-proxy file we can use when we can deploy proxy + init container, without the manager agent.

File: test/integration/mixer.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        infra: mixer
+    spec:
+      containers:
+      - name: mixer
+        image: {{.hub}}/mixer
```
Comment (kyessenov) [inline test/integration/mixer.yaml.tmpl]: let's use mixer SHA as a tag here, so we can identify which mixer build it is.

File: test/integration/mixer.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: mixer
+  labels:
+    infra: mixer
+spec:
+  ports:
+  - port: 9091
+    name: tcp
```
Comment (kyessenov) [inline test/integration/mixer.yaml.tmpl]: nit: it's TCP by default. "grpc" is more appropriate.

File: test/integration/manager-discovery.yaml.tmpl [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        infra: manager
+    spec:
+      containers:
+      - name: manager
+        image: {{.hub}}/runtime:{{.tag}}
+        args: ["discovery", "--logtostderr", "-v", "4"]
```
Comment (kyessenov) [inline test/integration/manager-discovery.yaml.tmpl]: -v 4 triggers lots of noise form k8s client libraries. maybe keep it -v 2?

File: test/integration/BUILD [hunk 6]
```diff
@@ -19,10 +19,11 @@ go_library(
 go_binary(
     name = "integration",
     data = [
-        "external-services.yaml.tmpl",
-        "http-service.yaml.tmpl",
-        "manager.yaml.tmpl",
```
Comment (kyessenov) [inline test/integration/BUILD]: is this file still around? if it is, please delete it.

File: test/integration/driver.go [hunk 7]
```diff
@@ -121,9 +123,20 @@ func setup() {
 
 	pods = make(map[string]string)
 
-	check(setupManager())
-	check(setupSimpleApp())
-	check(setupVersionedApp())
+	// deploy istio-infra
+	check(deploy("http-discovery", "http-discovery", managerDiscovery, namespace,
+		"8080", "80", "unversioned"))
```
Comment (kyessenov) [inline test/integration/driver.go]: nit: I would not specify values that are not used (like ports)

**Comment by kyessenov** [comment]
Feel free to merge.
